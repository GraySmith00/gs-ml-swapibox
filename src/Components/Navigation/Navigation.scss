.header {
  background-color: rgba(0, 0, 0, 0.5);
  padding: 20px 0 20px 0;
  position: fixed;
  top: 0;
  width: 100%;
  font-size: 1.5em;
  color: #fff;
}

.header-container {
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 80%;
  margin: 0 auto;
}

.brand-image {
  height: 1em;
}

.nav-btns {
  display: flex;
  justify-content: space-between;
  width: 50%;
}

.nav-link {
  text-decoration: none;
  margin-left: 1em;
  color: #fff;

  &:hover {
    transform: scale(1.1);
    transition: ease-in-out 0.3s;
  }
}

@media screen and (max-width: 900px) {
  .header-container,
  .main-container {
    width: 90%;
  }
}

@media screen and (max-width: 815px) {
  .header-container {
    flex-direction: column;
    width: 95%;
  }

  .brand-image {
    margin-bottom: 1rem;
  }

  .nav-btns {
    width: 75%;
  }
}

@media screen and (max-width: 515px) {
  .nav-btns {
    width: 95%;
  }

  .nav-link {
    margin-left: 0;
  }
}
