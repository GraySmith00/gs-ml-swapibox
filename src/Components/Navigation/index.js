import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Navigation = ({ favoritesData }) => {
  return (
    <React.Fragment>
      <NavLink exact to="/" className="nav-link">
        <img
          className="brand-image"
          src={require('../../images/starWarsLogoOneLine.png')}
          alt="star wars logo"
        />
      </NavLink>
      <nav className="nav-btns">
        <NavLink
          exact
          to="/people"
          className="nav-link"
          activeStyle={{
            color: '#feda4a'
          }}
        >
          People
        </NavLink>
        <NavLink
          exact
          to="/planets"
          className="nav-link"
          activeStyle={{
            color: '#feda4a'
          }}
        >
          Planets
        </NavLink>
        <NavLink
          exact
          to="/vehicles"
          className="nav-link"
          activeStyle={{
            color: '#feda4a'
          }}
        >
          Vehicles
        </NavLink>
        <NavLink
          exact
          to="/favorites"
          className="nav-link"
          activeStyle={{
            color: '#feda4a'
          }}
        >
          {`Favorites (${favoritesData.length})`}
        </NavLink>
      </nav>
    </React.Fragment>
  );
};

Navigation.propTypes = {
  favoritesData: PropTypes.array.isRequired
};

export default Navigation;
