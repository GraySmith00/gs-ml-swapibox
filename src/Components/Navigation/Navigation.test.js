import React from 'react';
import { shallow } from 'enzyme';

import Navigation from '../Navigation';

describe('Navigation component', () => {
  let wrapper;
  let mockFavoritesData;

  beforeEach(() => {
    mockFavoritesData = [
      {
        favorite: true,
        name: 'Luke Skywalker',
        planet: 'Tatooine',
        population: '200000',
        species: 'Human'
      },
      {
        favorite: false,
        name: 'C-3PO',
        planet: 'Tatooine',
        population: '200000',
        species: 'Droid'
      }
    ];
    wrapper = shallow(<Navigation favoritesData={mockFavoritesData} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
