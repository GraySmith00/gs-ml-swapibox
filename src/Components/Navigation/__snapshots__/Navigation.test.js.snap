// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    favoritesData={
        Array [
            Object {
              "favorite": true,
              "name": "Luke Skywalker",
              "planet": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "favorite": false,
              "name": "C-3PO",
              "planet": "Tatooine",
              "population": "200000",
              "species": "Droid",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/"
>
          <img
                    alt="star wars logo"
                    className="brand-image"
                    src="starWarsLogoOneLine.png"
          />
</NavLink>,
        <nav
          className="nav-btns"
>
          <NavLink
                    activeClassName="active"
                    activeStyle={
                              Object {
                                        "color": "#feda4a",
                                      }
                    }
                    aria-current="page"
                    className="nav-link"
                    exact={true}
                    to="/people"
          >
                    People
          </NavLink>
          <NavLink
                    activeClassName="active"
                    activeStyle={
                              Object {
                                        "color": "#feda4a",
                                      }
                    }
                    aria-current="page"
                    className="nav-link"
                    exact={true}
                    to="/planets"
          >
                    Planets
          </NavLink>
          <NavLink
                    activeClassName="active"
                    activeStyle={
                              Object {
                                        "color": "#feda4a",
                                      }
                    }
                    aria-current="page"
                    className="nav-link"
                    exact={true}
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
          <NavLink
                    activeClassName="active"
                    activeStyle={
                              Object {
                                        "color": "#feda4a",
                                      }
                    }
                    aria-current="page"
                    className="nav-link"
                    exact={true}
                    to="/favorites"
          >
                    Favorites (2)
          </NavLink>
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt="star wars logo"
            className="brand-image"
            src="starWarsLogoOneLine.png"
/>,
          "className": "nav-link",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "star wars logo",
            "className": "brand-image",
            "src": "starWarsLogoOneLine.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              activeStyle={
                            Object {
                                          "color": "#feda4a",
                                        }
              }
              aria-current="page"
              className="nav-link"
              exact={true}
              to="/people"
>
              People
</NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                            Object {
                                          "color": "#feda4a",
                                        }
              }
              aria-current="page"
              className="nav-link"
              exact={true}
              to="/planets"
>
              Planets
</NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                            Object {
                                          "color": "#feda4a",
                                        }
              }
              aria-current="page"
              className="nav-link"
              exact={true}
              to="/vehicles"
>
              Vehicles
</NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                            Object {
                                          "color": "#feda4a",
                                        }
              }
              aria-current="page"
              className="nav-link"
              exact={true}
              to="/favorites"
>
              Favorites (2)
</NavLink>,
          ],
          "className": "nav-btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#feda4a",
              },
              "aria-current": "page",
              "children": "People",
              "className": "nav-link",
              "exact": true,
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#feda4a",
              },
              "aria-current": "page",
              "children": "Planets",
              "className": "nav-link",
              "exact": true,
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#feda4a",
              },
              "aria-current": "page",
              "children": "Vehicles",
              "className": "nav-link",
              "exact": true,
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#feda4a",
              },
              "aria-current": "page",
              "children": "Favorites (2)",
              "className": "nav-link",
              "exact": true,
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites (2)",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            exact={true}
            to="/"
>
            <img
                        alt="star wars logo"
                        className="brand-image"
                        src="starWarsLogoOneLine.png"
            />
</NavLink>,
          <nav
            className="nav-btns"
>
            <NavLink
                        activeClassName="active"
                        activeStyle={
                                    Object {
                                                "color": "#feda4a",
                                              }
                        }
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/people"
            >
                        People
            </NavLink>
            <NavLink
                        activeClassName="active"
                        activeStyle={
                                    Object {
                                                "color": "#feda4a",
                                              }
                        }
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/planets"
            >
                        Planets
            </NavLink>
            <NavLink
                        activeClassName="active"
                        activeStyle={
                                    Object {
                                                "color": "#feda4a",
                                              }
                        }
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
            <NavLink
                        activeClassName="active"
                        activeStyle={
                                    Object {
                                                "color": "#feda4a",
                                              }
                        }
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/favorites"
            >
                        Favorites (2)
            </NavLink>
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="star wars logo"
              className="brand-image"
              src="starWarsLogoOneLine.png"
/>,
            "className": "nav-link",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "star wars logo",
              "className": "brand-image",
              "src": "starWarsLogoOneLine.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                activeStyle={
                                Object {
                                                "color": "#feda4a",
                                              }
                }
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/people"
>
                People
</NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                                Object {
                                                "color": "#feda4a",
                                              }
                }
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/planets"
>
                Planets
</NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                                Object {
                                                "color": "#feda4a",
                                              }
                }
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/vehicles"
>
                Vehicles
</NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                                Object {
                                                "color": "#feda4a",
                                              }
                }
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/favorites"
>
                Favorites (2)
</NavLink>,
            ],
            "className": "nav-btns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#feda4a",
                },
                "aria-current": "page",
                "children": "People",
                "className": "nav-link",
                "exact": true,
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#feda4a",
                },
                "aria-current": "page",
                "children": "Planets",
                "className": "nav-link",
                "exact": true,
                "to": "/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#feda4a",
                },
                "aria-current": "page",
                "children": "Vehicles",
                "className": "nav-link",
                "exact": true,
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#feda4a",
                },
                "aria-current": "page",
                "children": "Favorites (2)",
                "className": "nav-link",
                "exact": true,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites (2)",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
